pymodbus/client/async/asyncio/__init__.py

```diff
@@ -718,8 +725,8 @@ def protocol_made_connection(self, protocol):
         Protocol notification of successful connection.
         """
         _logger.info('Protocol made connection.')
-        if not self._connected:
-            self._connected = True
+        if not self._connected_event.is_set():
```

Comment:
You can directly use the property `_connected` here.

Reply:
I didn't want to break compatibility with possible subclasses or mixins. So i created the `_connected` property that returns the state of the connected event. But you are right it is cleaner to use the `_property` inside the class as well. I will update the pull request.